boot/freeldr/freeldr/arch/uefi/uefildr.c

```diff
@@ -25,12 +30,50 @@ EfiEntry(
     GlobalImageHandle = ImageHandle;
     GlobalSystemTable = SystemTable;
 
-    BootMain(NULL);
+    /* Debugger pre-initialization */
+    DebugInit(0);
+
+    MachInit("");
+
+    /* UI pre-initialization */
+    if (!UiInitialize(FALSE))
+    {
+        UiMessageBoxCritical("Unable to initialize UI.");
+        //goto Quit;
```

Comment:
```suggestion
        //goto Quit;
```
why leaving that commented out code inside?
If it's needed, uncomment it.
**If it's not needed, strip it.**
If it would be needed actually, but causes issues due to bugs otherwhere, please add a comment about what would go wrong if it was enabled and where you believe a fix should go.